src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TabKey.cs

```diff
@@ -67,60 +64,91 @@ private bool TryInvokeSnippetCompletion(TabKeyCommandArgs args, CompletionServic
 
             var text = subjectBuffer.AsTextContainer().CurrentText;
 
-            // Delete the ? and invoke completion
+            // If the user types "<line start><spaces><question><tab>"
```

Comment:
Sorry, you're right, forgot about that heuristic.

So to summarize for my own sanity, `<whitespace><question><tab>` _always_ brings up the picker, no matter what's before it, and `<identifier><question><tab>` _only_ brings up our special picker if it's the first identifier on the line (as a heuristic because most-if-not-all snippets are statements, not expressions).